### ISSUE TITLE
subset by cycle for "argos" type not working properly

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#402]
I again found this as a result of the development of `qcApp()`. The following code demonstrates the problem:
```
data(index)
index1 <- subset(index, 1:30)
index2 <- subset(index1, ID="1901584")
argos2 <- readProfiles(getProfiles(index2))
argos3 <- subset(argos2, cycle="124")
```
Gives:
```R
Kept 1 profiles 
```
However, if we do:

```
argos3[["cycles"]]
```
The following occurs:

```R
[1] "124" "125" "126" "128" "129" "131" "132" "147"
 [9] "148" "150"
```
And we see it didn't actually subset

**Comment by dankelley** [comment]
Who is assigned this?  (reprex?) (z?)

**Comment by dankelley** [comment]
This is fixed in "develop" commit 8883a840c63ba5b36f3daae640f2aa3a25efc038.  The error was in using `keep` on `data@index`, which must have resulted from a copy/paste from similar actions for objects of type `index`.

<details><summary>reprex</summary>


``` r
library(argoFloats)
data(index)
index1 <- subset(index, 1:30)
#> Kept 30 profiles (2.98%)
index2 <- subset(index1, ID="1901584")
#> Kept 10 profiles (33.3%)
argos2 <- readProfiles(getProfiles(index2))
#> Warning in readProfiles(getProfiles(index2)): Of 10 profiles read, 9 have >10% of salinity values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     1 2 3 4 5 6 7 8 10
#> Warning in readProfiles(getProfiles(index2)): Of 10 profiles read, 9 have >10% of temperature values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     1 2 3 4 5 6 7 8 10
argos3 <- subset(argos2, cycle="124")
#> Kept 1 profiles
stopifnot(1 == argos3[["length"]])
```

<sup>Created on 2021-02-18 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by j-harbin** [comment]
Thanks, Dan! I can confirm this works on my end as well. 

PS- I promise eventually I will get used to reprex..... :)
